src/renderers/shared/reconciler/ReactCompositeComponent.js

```diff
@@ -808,8 +808,10 @@ var ReactCompositeComponentMixin = {
   attachRef: function(ref, component) {
     var inst = this.getPublicInstance();
     invariant(inst != null, 'Stateless function components cannot have refs.');
+    var componentInstance = component.getPublicInstance();
```

Comment:
Dig. I definitely punted on what to name that. Will go with `publicComponentInstance`.